Rewrite mat reshape transpose testers

paulinagacek opened the pull request:
### PR types
Others

### PR changes
Others

### Describe
Deleted old C++ scale matmul fuse pass test and updated python one

Currently in PaddlePaddle there are 3 unit-test frameworks - one written in C++ and 2 in Python.
Old fuse pass tests (C++) fuse_pass_tester.cc are checking only if graph has correct number of / type of vertices after fusion. It doesn't check data. So these tests pass even if accuracy drops.
New python framework is superior because it checks both graph fusion and accuracy after fusion.

Tests rewritten in this PR:

- reshape_transpose_matmul_mkldnn_fuse_pass_tester.cc
- matmul_transpose_reshape_mkldnn_fuse_pass_tester.cc

Commit:
reshape_transpose_matmul_pass_tester rewritten

Commit:
matmul_transpose_reshape_pass_tester rewritten

paulinagacek commented:
@Silv3S @HulekJakub could you review please? ^^

Silv3S commented on python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_reshape_transpose_matmul_fuse_pass.py:
```diff
@@ -20,10 +20,11 @@
-num = 32 * 64
-
 
 class TestReshapeTransposeMatmulMkldnnFusePass(PassAutoScanTest):
```
Could you please rename filenames and all occurrences of `MKLDNN` to `OneDNN`? I know that fuse pass name is still mkldnn, but tests can be already renamed.

paulinagacek replied:
sure

Silv3S reviewed (dismissed):
LGTM

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_mat_reshape_transpose_testers

Commit:
mkldnn to onednn

paulinagacek commented:
@kolinwei @XieYunshen please review and approve changes in unnitests. I deleted c++ tests becuase they were no longer needed (more about it in description above) and renamed python UT to match new convention.

HulekJakub reviewed (approved):
LGTM

yaomichael commented:
@yeliang2258 please help find someone to approve this.
